src/ngraph_builder.cc

```diff
@@ -2086,6 +2086,212 @@ static Status TranslateReciprocalOp(
       });
 }
 
+static void ComputeScaleOffsetFolded(const uint& num_bits,
+                                     const bool& unsigned_type,
+                                     const bool& scaled, const int min_range,
+                                     const int max_range, float* scale,
+                                     int* offset) {
+  int scaled_elide = scaled ? 1 : 0;
+  int min_type = 0;
+  if (!unsigned_type) min_type = -((2 ^ (num_bits - 1)) - scaled_elide);
+  uint raise_to = num_bits - (unsigned_type ? 0 : 1);
+  int max_type = (2 ^ raise_to) - 1 - scaled_elide;
+  float adj_min_range, adj_max_range;
+  if (scaled) {
+    auto abs_min_range = std::abs(min_range);
+    auto abs_max_range = std::abs(max_range);
+    auto range_boundary = std::max(abs_min_range, abs_max_range);
+    adj_min_range = -range_boundary;
+    adj_max_range = range_boundary;
+  } else {
+    // TODO: Adjust range or fail?
+    adj_min_range = std::min(min_range, 0);
+    adj_max_range = std::max(max_range, 0);
+  }
+  *scale = (adj_max_range - adj_min_range) / (max_type - min_type);
+  // TODO: should it be: round(adj_min_range / *scale) (or floor)?
+  *offset = min_type - std::lround(adj_min_range / *scale);
+}
+
```

Comment:
currently both TranslateQuantizeV2Op and TranslateDequantizeOp assume static input and we compute the scale-offset into ng_consts. In the future we might need to support converting min-max to scale-offset using a ngraph subgraph

Reply:
In that case may be put a comment with TODO and mentioning this.